src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py

```diff
@@ -701,21 +818,35 @@ def predict(self, data, verbose = True):
             Rows: 10
             [3, 4, 3, 3, 4, 5, 8, 8, 8, 4]
         """
+        output_type = output_type.lower()
+        _tkutl._check_categorical_option_type("output_type", output_type, 
+            ["probability", "class", "probability_vector"])
         if isinstance(data, _tc.SArray):
             predicted = self._predict_with_probabilities(
                 _tc.SFrame({
                     self.feature: data
                 }),
+                batch_size,
                 verbose
             )
         elif isinstance(data, _tc.SFrame):
-            predicted = self._predict_with_probabilities(data, verbose)
+            predicted = self._predict_with_probabilities(data, batch_size, verbose)
         else:
             # single input
             predicted = self._predict_with_probabilities(
                 _tc.SFrame({
                     self.feature: [data]
                 }),
+                batch_size,
                 verbose
             )
-        return predicted[self.target]
+        if output_type == "class":
+            return predicted[self.target]
+        elif output_type == "probability":
+            _class_to_index = self._class_to_index
+            target = self.target
+            return predicted.apply(
```

Comment:
An `apply` on just the SArray probably has better performance and I think more readable.

Reply:
But I need to be able to access both columns in the SFrame, i.e. `target` and `probability`.

Reply:
ah, I missed the second `row` in the lambda expression.